spec/defines/sslcertificate_spec.rb

```diff
@@ -60,46 +65,45 @@
   #  }}
   #
   #  it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass password to Sslcertificate\[certificate-testCert\]/) }
-  #end
+  # end
 
   describe 'when no certificate location is provided' do
     let(:title) { 'certificate-testCert' }
     let(:params) { {
-        :name       => 'testCert',
-        :password   => 'testPass',
-        :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
-        :root_store => 'LocalMachine',
-        :store_dir  => 'My',
+      :name       => 'testCert',
+      :password   => 'testPass',
+      :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
+      :root_store => 'LocalMachine',
+      :store_dir  => 'My',
     }}
 
-    it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass location to Sslcertificate\[certificate-testCert\]/) }
+    it { expect { should contain_exec('Install-SSL-Certificate-testCert') }.to raise_error(Puppet::Error) }
   end
 
   describe 'when empty certificate location is provided' do
     let(:title) { 'certificate-testCert' }
     let(:params) { {
-        :name       => 'testCert',
-        :password   => 'testPass',
-        :location   => '',
-        :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
-        :root_store => 'LocalMachine',
-        :store_dir  => 'My',
+      :name       => 'testCert',
+      :password   => 'testPass',
+      :location   => '',
+      :thumbprint => '07E5C1AF7F5223CB975CC29B5455642F5570798B',
+      :root_store => 'LocalMachine',
+      :store_dir  => 'My',
     }}
 
-    it { expect { should contain_exec('Install-SSL-Certificate-testCert')}.to raise_error(Puppet::Error, /Must pass location to sslcertificate/) }
+    it { expect { should contain_exec('Install-SSL-Certificate-testCert') }.to raise_error(Puppet::Error, /Must pass location to sslcertificate/) }
```

Comment:
The ```let params``` in the RSpec is an empty string, (a value has been provided for a mandatory variable) hence the custom failure message in ```validate_re``` will kick in